homeassistant/components/traccar/device_tracker.py

```diff
@@ -172,11 +159,24 @@ def __init__(self, api, hass, async_see, scan_interval,
             for custom_attr in self._custom_attributes:
                 if device_info.get(custom_attr) is not None:
                     attr[custom_attr] = device_info[custom_attr]
+                    if custom_attr in self._skip_accuracy_on:
+                        skip_accuracy_filter = True
+
+            accuracy = float(0)
+            if device_info.get('accuracy') is not None:
+                accuracy = device_info['accuracy']
+            if (not skip_accuracy_filter and self._max_accuracy > 0 and
+                    accuracy > self._max_accuracy):
+                _LOGGER.info('Excluded position by accuracy filter: %f (%s)',
+                             accuracy, attr[ATTR_TRACCAR_ID])
+                continue
+
             await self._async_see(
                 dev_id=slugify(device_info['device_id']),
                 gps=(device_info.get('latitude'),
                      device_info.get('longitude')),
-                gps_accuracy=(device_info.get('accuracy')),
+                gps_accuracy=accuracy,
+                battery=device_info.get('battery'),
```

Comment:
Would this result in a breaking change for users that have set up templates to get the battery?

Reply:
Oh yes! this is a breaking change for these users since [160](https://github.com/home-assistant/home-assistant/pull/23292/files#diff-5192f6e34a898c79e0b402348c7a7a65L160) and [161](https://github.com/home-assistant/home-assistant/pull/23292/files#diff-5192f6e34a898c79e0b402348c7a7a65L161) lines has been deleted from actual version.
I think this is the right way to push battery info in the device_tracker, it's expected on async_see method so I don't know the reason to create a new custom attribute for the device_tracker.

With this modification device_tracker attribute named battery_level in the past now is named battery. I will document it in the PR.

Reply:
Can you add some more description to the breaking change note?:
- Why it's needed
- What the user needs to do

Reply:
Please, check if it's better now :)

Reply:
Much better :)